pkg/probe/probes.go

```diff
@@ -136,19 +136,27 @@ func checkNodeReadiness(cli client.Client) (bool, error) {
 }
 
 func defaultGw(currentState gjson.Result) (string, error) {
-	var found gjson.Result
+	var found string
 	currentState.Get("routes.running").ForEach(
 		func(_, v gjson.Result) bool {
 			// we want to pick the next hop related to the "main" table because we may have multiple tables
-			if v.Get("destination").String() == "0.0.0.0/0" && v.Get("table-id").Int() == mainRoutingTableID {
-				found = v.Get("next-hop-address")
+			if (v.Get("destination").String() == "0.0.0.0/0" || v.Get("destination").String() == "::/0") &&
+				v.Get("table-id").Int() == mainRoutingTableID {
+				// if next hop is IPv6 link-local, we need to append an interface otherwise it is
+				// not clear which interface should be used for communication (e.g. ping test)
+				found = v.Get("next-hop-address").String()
+				ip := net.ParseIP(found)
+				if ip.To4() == nil && ip.IsLinkLocalUnicast() {
+					iface := v.Get("next-hop-interface").String()
+					found = found + "%" + iface
+				}
```

Comment:
If this works for ipv4 + ipv6 not link-local we can remove the condition to simplify code

```suggestion
				found = v.Get("next-hop-address").String()
				ip := net.ParseIP(found)
				iface := v.Get("next-hop-interface").String()
				found = found + "%" + iface
				
```

Also an extra-mile would be to return the "Route" struct instead of a string and consume it at `runPing` so the `found + "%" + iface`  composition is done there, I think is something specific of `ping` command

Reply:
That's a good point, strings are ugly so I redesigned it to a struct `Route` that holds IP address and interface